### ISSUE TITLE
PnP questions

### ISSUE THREAD

**Issue by kaixinbear** [tjiiv-cprg/MonoRUn#2]
Hi, Lakonik~ Thansk for your great work. I wanna to know you two questions.

1. It seems that the code base you mean is not corresponding to the paper?
Leveraging the differentiable PnP, we apply smooth L1 loss on the Euclidean errors of estimated translation vector t∗ and yaw angle β∗

2. If you don't include differentiable PnP in the code, how does the gradient of object pose be backprop in network ? The gradient of translation could not be propagated to the network in this code base?
Thanks.

**Comment by Lakonik** [comment]
Hi! Thank you for your interest in our work.

1. As mentioned in the paper, training with differentiable PnP is not our main idea and is only investigated in ablation studies, where it shows adverse effect on the network trained with Robust KL loss. Therefore, differentiable PnP is not included in the final version, so we only released the code for the two setups: "Fully Self-Supervised Reconstruction (Without Extra Supervision)" & "LiDAR Supervision".

2. Without PnP back-propagation, we mainly use the reconstruction loss for reprojected crds. (Robust KL loss) as a surrogate pose loss. There is no gradient of translation in network, but reprojection will do the same job.

Feel free for any further questions.
